protocols/raft/src/main/java/io/atomix/protocols/raft/RaftFailureListener.java

```diff
@@ -0,0 +1,8 @@
+package io.atomix.protocols.raft;
+
+import java.util.concurrent.CompletableFuture;
+
+@FunctionalInterface
+public interface RaftFailureListener {
+   CompletableFuture<Void> onRaftFailed();
```

Comment:
As with any other public interfaces, can we document it such that people know how to implement it in the future?